lib/puppet/settings/server_list_setting.rb

```diff
@@ -0,0 +1,20 @@
+class Puppet::Settings::ServerListSetting < Puppet::Settings::ArraySetting
+
+  def type
+    :server_list
+  end
+
+  def munge(value)
+    servers = super 
+    servers.map! { |server| 
+      case server
+      when String
+        server.split(':')
```

Comment:
the `super` call handles comma-sepration via the `array` class - this is splitting each `server:port` pair into their array, so a string of the form `"foo:123,bar:456"` becomes `[["foo",123],["bar",456]]`